### PR TITLE
GITHUB#13175: Stop double-checking priority queue inserts in some FacetCount classes

### LINKED ISSUE #13175: Stop double-checking priority queue inserts
### Description

There are several occurrences in our code where we check that an incoming value can be inserted in a priority queue and then call `insertWithOverflow`, which re-does the check. Let's stop checking before calling `insertWithOverflow`.

Credit to @epotyom for [noticing this in `AbstractSortedSetDocValueFacetCounts`](https://github.com/apache/lucene/pull/12966#discussion_r1457584536).

### PR DESCRIPTION
**Pull request by slow-J**
Removing 2 cases of bottomX optimizations where insertWithOverflow already handles the check.

Closes #13175 

The diff looks funny because of unnesting, but, per file, I am just removing `bottomOrd` and `bottomCount` and removing an if statement.

### CODE REVIEW

File: lucene/CHANGES.txt [hunk 1]
```diff
@@ -273,6 +273,8 @@ Optimizations
 
 * GITHUB#13431: Replace TreeMap and use compiled Patterns in Japanese UserDictionary. (Bruno Roustant)
 
+* GITHUB#13175: Stop double-checking priority queue inserts. (Jakub Slowinski)
```
Comment (mikemccand) [inline lucene/CHANGES.txt:276]: Maybe clarify that this just applies to some of our facet counting impls (not to the many other PQs Lucene has/uses)?

**Review by mikemccand — APPROVED** [review]

File: lucene/CHANGES.txt (hunk 1 above)
Comment (slow-J) [reply lucene/CHANGES.txt:276]: Good point Mike, will do.

**Comment by slow-J** [comment]
Hi @mikemccand can we merge or are we waiting for more reviewers.

**Comment by mikemccand** [comment]
Thanks @slow-J -- I just backported to 9.12 as well.  I had to resolve a few conflicts, maybe have a peek and see if I did it correctly?

**Comment by slow-J** [comment]
Thanks @mikemccand I think the diff for 9.12 looks good.

**Pull request merged**